llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -310,6 +310,15 @@ std::string saveModuleProperties(module_split::ModuleDesc &MD,
   auto PropSet =
       computeModuleProperties(MD.getModule(), MD.entries(), GlobProps);
 
+  // When the split mode is none, the required work group size will be added
+  // to the whole module, which will make the runtime unable to
+  // launch the other kernels in the module that have different
+  // required work group sizes or no requried work group sizes. So we need to
+  // remove the required work group size metadata in this case.
+  if (SplitMode == module_split::SPLIT_NONE)
+    PropSet.remove(PropSetRegTy::SYCL_DEVICE_REQUIREMENTS,
+                   "reqd_work_group_size_uint64_t");
```

Comment:
should we make `reqd_work_group_size_uint64_t` a constant somewhere so we dont have any typo problems in the future?